manager/sql/manager_sql.go

```diff
@@ -267,28 +160,11 @@ func (s *SQLManager) create(policy Policy, tx *sqlx.Tx) (err error) {
 				return errors.WithStack(err)
 			}
 
-			switch s.db.DriverName() {
-			case "postgres", "pgx":
-				if _, err := tx.Exec(s.db.Rebind(fmt.Sprintf("INSERT INTO ladon_%s (id, template, compiled, has_regex) SELECT ?, ?, ?, ? WHERE NOT EXISTS (SELECT 1 FROM ladon_%[1]s WHERE id = ?)", v.t)), id, template, compiled.String(), strings.Index(template, string(policy.GetStartDelimiter())) > -1, id); err != nil {
-					return errors.WithStack(err)
-				}
-
-				if _, err := tx.Exec(s.db.Rebind(fmt.Sprintf("INSERT INTO ladon_policy_%s_rel (policy, %[1]s) SELECT ?, ? WHERE NOT EXISTS (SELECT 1 FROM ladon_policy_%[1]s_rel WHERE policy = ? AND %[1]s = ?)", v.t)), policy.GetID(), id, policy.GetID(), id); err != nil {
-					return errors.WithStack(err)
-				}
-				break
-
-			case "mysql":
-				if _, err := tx.Exec(s.db.Rebind(fmt.Sprintf("INSERT IGNORE INTO ladon_%s (id, template, compiled, has_regex) VALUES (?, ?, ?, ?)", v.t)), id, template, compiled.String(), strings.Index(template, string(policy.GetStartDelimiter())) > -1); err != nil {
-					return errors.WithStack(err)
-				}
-
-				if _, err := tx.Exec(s.db.Rebind(fmt.Sprintf("INSERT IGNORE INTO ladon_policy_%s_rel (policy, %s) VALUES (?, ?)", v.t, v.t)), policy.GetID(), id); err != nil {
-					return errors.WithStack(err)
-				}
-				break
-			default:
-				return errors.Errorf("Database driver %s is not supported", s.db.DriverName())
+			if _, err := tx.Exec(query, id, template, compiled.String(), strings.Index(template, string(policy.GetStartDelimiter())) >= -1); err != nil {
+				return errors.WithStack(err)
+			}
+			if _, err := tx.Exec(queryRel, policy.GetID(), id); err != nil {
```

Comment:
By using `db.Rebind()` we can actually further streamline the SQL statements and use `?` everywhere instead of dialect-specific parameters.